enhancements/installer/signal-cluster-deletion.md

```diff
@@ -0,0 +1,165 @@
+
+### Non-Goals
+
+* Teardown of the actual cluster itself. (control plane, instances, etc) This process is expected to still be run external to the cluster.
+* Attaching finalizers to operator resources based on the `Alive` object which would facilitate removal of operator resources.
+* Removing resources other than the `Alive` object during cluster deprovision.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As an OpenShift administrator, I would like for any external resources created in service of the cluster be removed when when I delete the cluster, so that:
+* I do not have to track down those resources and remove them manually
+* I do not have to pay for resources that are not being used.
+
+#### Story 2
+As an OpenShift (software, big-O) Operator, I would like to have a way to know that the cluster is being shut down, so that I can perform any necessary cleanup steps and remove external resources.
+
+#### Story 3
+As an OpenShift administrator, I would like to have a way to indicate that certain external resources that would normally be cleaned up on cluster deletion be preserved on cluster delete, so that I don't lose important resources stored externally.
+
+### Implementation Details/Notes/Constraints
+
+#### API
+
+Introduce a new CRD `Alive` from which operator resources could attach a finalizer initiating cascading delete when the `Alive` resource is deleted.
+
+```yaml
+apiVersion: v1alpha1
+kind: Alive
+metadata:
+  name: cluster
+  namespace: openshift-config
+spec:
+```
+
+*Note*: We proposed adding the CRD and admission plugin (which pevents delete) to the cluster version operator and it was decided that they would be better placed within OLM. We would prefer not to create a separate operator for a CRD and admission plugin as that would require a repository and release engineering. We're requesting feedback on where to put these resources.
+
+#### Provision
+
+Create an `Alive` object during installation.
+
+How this happens remains unknown. We do not wish to create an entire second level operator just to deploy this resource. We do not wish to stuff it into the CVO. Some kind of mechanism to a CRD and CR manifest into the release repo seems most likley.
+
+#### Upgrade
+
+Create an `Alive` object for existing clusters.
+
+If the resource is carried in the release image, this should handle the upgrade case.
+
+#### Deprovision
+
+Best practice for entities that provision/deprovision clusters (openshift-install, Hive, managed offerings from cloud partners) would be to make use of this deletion to allow in-cluster resources to clean up. However we accept that this is a best effort, opt-in solution, and not all implementations will make use of it. (leaving them no worse off than they are today)
+
+The best practice would then become:
+
+  1. Delete the Alive custom resource before destroying the cluster itself.
```

Comment:
At which point of cluster de-provision process is `Alive` deleted?

Volumes in some (most?) cloud platforms can't be deleted until all pods that use the volume are deleted and the volumes are unmounted + detached from nodes. Should a new storage operator find these pods and start killing them in order to remove all storage resources when `Alive` is marked for deletion? The pods are likely to be recreated by StatefulSets or even other operators outside of our control. Or can we expect that `Alive` is deleted when all user workloads (Pods, StatefulSets, ...) or even Nodes are already deleted? What if a cluster component uses storage that needs to be cleaned out (e.g. monitoring or image registry)?